File: sample/src/main/java/com/github/arturogutierrez/sample/MainActivity.java
package com.github.arturogutierrez.sample;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

import com.github.arturogutierrez.Badges;
import com.github.arturogutierrez.BadgesNotSupportedException;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = MainActivity.class.getSimpleName();

    EditText badgeCountText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        badgeCountText = findViewById(R.id.text_badge_count);
    }

    public void updateButtonPressed(View v) {
        int badgeCount = Integer.parseInt(badgeCountText.getText().toString());
        updateBadge(badgeCount);
    }

    private void updateBadge(int count) {
        try {
            Badges.setBadge(this, count);
        } catch (BadgesNotSupportedException badgesNotSupportedException) {
            Log.d(TAG, badgesNotSupportedException.getMessage());
        }
    }
}

File: sample/src/main/java/com/github/arturogutierrez/sample/MainActivity.kt
package com.github.arturogutierrez.sample

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.github.arturogutierrez.Badges
import com.github.arturogutierrez.BadgesNotSupportedException

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.buttonUpdate).setOnClickListener { updateBadgeCount() }
    }

    private fun updateBadgeCount() {
        val badgeCount =
            findViewById<TextView>(R.id.editTextBadgeCount).text.toString().toIntOrNull()
        badgeCount?.let {
            try {
                Badges.setBadgeCount(this, badgeCount)
            } catch (badgesNotSupportedException: BadgesNotSupportedException) {
                Log.d(MainActivity::class.java.simpleName, badgesNotSupportedException.message!!)
            }
        }
    }

}